iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RevertConnectVoltageLevelOnLine.java

```diff
@@ -0,0 +1,196 @@
+                return;
+            }
+        }
+
+        VoltageLevel commonVl = network.getVoltageLevel(commonVlId);
+        Branch.Side line1Side1 = line1VlId1.equals(commonVlId) ? Branch.Side.TWO : Branch.Side.ONE;
+        Branch.Side line1Side2 = line1Side1 == Branch.Side.ONE ? Branch.Side.TWO : Branch.Side.ONE;
+        Branch.Side line2Side2 = line2VlId1.equals(commonVlId) ? Branch.Side.TWO : Branch.Side.ONE;
+        Branch.Side line2Side1 = line2Side2 == Branch.Side.ONE ? Branch.Side.TWO : Branch.Side.ONE;
+
+        // Set parameters of the new line replacing the two existing lines
+        LineAdder lineAdder = createLineAdder(lineId, lineName, line1Side1 == Branch.Side.TWO ? line1VlId2 : line1VlId1,
+                line2Side2 == Branch.Side.TWO ? line2VlId2 : line2VlId1, network, line1, line2);
+
+        attachLine(line1.getTerminal(line1Side1), lineAdder, (bus, adder) -> adder.setConnectableBus1(bus.getId()), (bus, adder) -> adder.setBus1(bus.getId()), (node, adder) -> adder.setNode1(node));
+        attachLine(line2.getTerminal(line2Side2), lineAdder, (bus, adder) -> adder.setConnectableBus2(bus.getId()), (bus, adder) -> adder.setBus2(bus.getId()), (node, adder) -> adder.setNode2(node));
+
+        // get line1 limits
+        LoadingLimitsBags limitsLine1Side1 = new LoadingLimitsBags(() -> line1.getActivePowerLimits(line1Side1), () -> line1.getApparentPowerLimits(line1Side1), () -> line1.getCurrentLimits(line1Side1));
+        LoadingLimitsBags limitsLine1Side2 = new LoadingLimitsBags(() -> line1.getActivePowerLimits(line1Side2), () -> line1.getApparentPowerLimits(line1Side2), () -> line1.getCurrentLimits(line1Side2));
+
+        // get line2 limits
+        LoadingLimitsBags limitsLine2Side1 = new LoadingLimitsBags(() -> line2.getActivePowerLimits(line2Side1), () -> line2.getApparentPowerLimits(line2Side1), () -> line2.getCurrentLimits(line2Side1));
+        LoadingLimitsBags limitsLine2Side2 = new LoadingLimitsBags(() -> line2.getActivePowerLimits(line2Side2), () -> line2.getApparentPowerLimits(line2Side2), () -> line2.getCurrentLimits(line2Side2));
+
+        // Remove the two existing lines
+        line1.remove();
+        removedLineReport(reporter, line1Id);
+
+        line2.remove();
+        removedLineReport(reporter, line2Id);
+
+        // Create the new line
+        Line line = lineAdder.add();
+        LoadingLimitsBags limitsSide1 = mergeLimits(line1Id, limitsLine1Side1, limitsLine1Side2, reporter);
+        LoadingLimitsBags limitsSide2 = mergeLimits(line2Id, limitsLine2Side2, limitsLine2Side1, reporter);
+
+        addLoadingLimits(line, limitsSide1, Branch.Side.ONE);
+        addLoadingLimits(line, limitsSide2, Branch.Side.TWO);
+        createdLineReport(reporter, lineId);
+        LOG.info("New line {} created, replacing lines {} and {}", lineId, line1Id, line2Id);
+
+        // remove voltage level and substation in common, if necessary
+        removeVoltageLevelAndSubstation(commonVl, reporter);
```

Comment:
About this method: could you add the messages in the Logger of `TopologyModifivationUtils` as well?

Reply:
Done